saleor/order/calculations.py

```diff
@@ -0,0 +1,268 @@
+from decimal import Decimal
+from typing import Iterable, List, Optional, Tuple
+
+from django.conf import settings
+from django.db.models import prefetch_related_objects
+from django.utils import timezone
+from prices import Money, TaxedMoney
+
+from ..core.prices import quantize_price
+from ..core.taxes import TaxData, TaxError, zero_taxed_money
+from ..plugins.manager import PluginsManager
+from . import ORDER_EDITABLE_STATUS
+from .interface import OrderTaxedPricesData
+from .models import Order, OrderLine
+
+
+def _apply_tax_data_from_manager(
+    manager: PluginsManager, order: Order, lines: Iterable[OrderLine]
+) -> bool:
+    currency = order.currency
+
+    try:
+        price_lines: List[dict] = []
+        for line in lines:
+            variant = line.variant
+            if not variant:
+                continue
+            product = variant.product
+
+            unit = manager.calculate_order_line_unit(order, line, variant, product)
+            total = manager.calculate_order_line_total(order, line, variant, product)
+            tax_rate = manager.get_order_line_tax_rate(
+                order, product, variant, None, unit.undiscounted_price
+            )
+
+            price_lines.append({"tax_rate": tax_rate, "unit": unit, "total": total})
+
+        subtotal = sum((line.total_price for line in lines), zero_taxed_money(currency))
+        shipping_price = manager.calculate_order_shipping(order)
+        shipping_tax_rate = manager.get_order_shipping_tax_rate(order, shipping_price)
+        total = shipping_price + subtotal
+    except TaxError:
+        return True
+    else:
+        order.total = total
+        order.shipping_price = shipping_price
+        order.shipping_tax_rate = shipping_tax_rate
+
+        for order_line, price_line in zip(lines, price_lines):
+            unit = price_line["unit"]
+            order_line.undiscounted_unit_price = unit.undiscounted_price
+            order_line.unit_price = unit.price_with_discounts
+
+            total = price_line["total"]
+            order_line.undiscounted_total_price = total.undiscounted_price
+            order_line.total_price = total.price_with_discounts
+
+            order_line.tax_rate = price_line["tax_rate"]
+
+        return False
+
+
+def _apply_tax_data(
+    order: Order, lines: Iterable[OrderLine], tax_data: TaxData
+) -> None:
+    def qp(net: Decimal, gross: Decimal) -> TaxedMoney:
```

Comment:
Maybe? This code should be more readable.
```suggestion
    def _quantize_price(net: Decimal, gross: Decimal) -> TaxedMoney:
```

Reply:
Resolved in recent commit.